### PR TITLE
[Heartbeat] Include cmd/status errs , add ECS Errs

### LINKED ISSUE #30170: [Heartbeat] Summary events override `cmd/status` events
Heartbeat's handling of `cmd/status` events https://github.com/elastic/beats/blob/master/x-pack/heartbeat/monitors/browser/synthexec/enrich.go#L136 is breaking the reporting of fatal script errors in the Uptime UI in that we no longer have `cmd/status` events sent to Uptime that can be used to trigger the view of stdout/stderr logs.

The behavior of the previously referenced code black ignores the `cmd/status` event in favor of creating a `heartbeat/summary` event. We should, instead, create a second `heartbeat/summary` event after the `cmd/status` event. This is a bit awkward in the current code structure since the enricher can only modify events, not create additional ones.

I'd recommend fixing this bug after we implement `discovery` mode, if we can transition things to the point where a single invocation of the synthetics node agent correlates to one journey that should simplify the code path.

### PR DESCRIPTION
**Pull request by andrewvc**
Fixes #30170 by taking a slightly different approach than previously thought.

We apparently suffered a regression in two places simultaneously:

1. Kibana stopped rendering components like [console output event list](https://github.com/elastic/kibana/blob/main/x-pack/plugins/synthetics/public/legacy_uptime/components/synthetics/console_output_event_list.tsx)
2. Heartbeat didn't really report these in a way that made sense in its schema, choosing between a final `cmd/status` and `summary` event.

This PR fixes the dilemma by always outputting a final summary event, and, in the case of an exit code `> 0` for the synthetics agent, adding a special error to that summary event. When we fix the UI regression we can look for that special error. That being said, I think this is somewhat overkill, for now, after investigating the Kibana UI further. Even with that being the case, I think this improves our error handling by bringing it more inline with ECS, so it's probably still worth merging as-is. See the detailed notes below for more info.

One issue we have here is that our error types are a bit of a mess. We have our standard `error.{type,message}` fields, which the synthetics agent traditionally did not use, in favor of its own `synthetics.error.{name,message,stack}` fields. 

Another problem is that the `error.type` field has always been quite coarse in heartbeat's usage, having only two values `validation` and `io`. One of our longer term aims is to integrate error codes for specific issues. This PR adds all of the ECS defined [error fields](https://www.elastic.co/guide/en/ecs/current/ecs-error.html) as optional in heartbeat. This lets us use a specific code to indicate a `cmd/status` error. See the new `ecserrs` package for more details there.

The implementation here is go-like, but perhaps not entirely intuitive. Heartbeat's continuation based job system has a signature like `type Job func (event) (conts []Job, err error)`, with a wrapper in `wrappers.go` actually filling in the top level `errors` JSON field at exactly one spot. So, we to make it possible to pack all the ECS error fields into a standard go error. This is done by implementing the `error` interface in the new `*ecserrs.ECSError` type, and using `errors.As` from the go stdlib to unpack ECS errors at serialization-time.

This PR also makes the `*SynthErr` type a superset of `*ecserrs.ECSError`, meaning that in the future we can just use ECS format in the synthetics agent for consistency.

A consequence of this work is that errors generated by synthetics will now be dual-indexed in the top level `error` field and the current `synthetics.error` field. Perhaps we can remove this dual storage in GA.

Given that there is no urgent bug here, and that this fix needs supporting Kibana changes to be useful, and that there's a decent amount of code churn here, I'm only targeting this at 8.4, not 8.3

This also moves our synthetic event type names over to constants, which is cleaner.

Finally, check the bottom of this ticket for some manual testing instructions. Ideally we'd have integration tests, but it's not easy to add those for browser monitors. This PR already makes a good number of improvements, and we can't justify the investment there right this minute.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

The following monitor (can be placed in `monitors.d` will the synthetics agent to exit uncleanly and trigger the status code error:

```yaml
- type: browser
  enabled: true
  id: browser-inline
  name: browser-inline
  playwright_options:
    ignoreHTTPSErrors: true
  throttling:
    download: 1.6
    upload: 0.75
    latency: 150
  source:
    inline:
      script:
        step("load homepage", async () => {
            await page.goto('https://www.elastic.co');
        });
        step("hover over products menu", async () => {
            await page.hover('css=[data-nav-item=products]');
        });
        step("failme", async () => {
            process.exit(123)
            await page.hhover('css=[data-nav-item=products]');
        });
  schedule: "@every 1m"
```

Produces the following in kibana

image

and

image

in discover:

image

### CODE REVIEW

File: x-pack/heartbeat/monitors/browser/synthexec/synthexec.go [hunk 1]
```diff
@@ -213,13 +214,12 @@ func runCmd(
 
 		var cmdError *SynthError = nil
 		if err != nil {
-			errMessage := fmt.Sprintf("command exited with status %d: %s", cmd.ProcessState.ExitCode(), err)
-			cmdError = &SynthError{Name: "cmdexit", Message: errMessage}
+			cmdError = ECSErrToSynthError(ecserr.NewBadCmdStatusErr(cmd.ProcessState.ExitCode(), loggableCmd.String()))
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/synthexec/synthexec.go:217]: The original error message is hidden here, Is this intended?

Comment (andrewvc) [reply x-pack/heartbeat/monitors/browser/synthexec/synthexec.go:217]: Yes! That error is just the string "exited with status 123", so it adds nothing.

**Review by vigneshshanmugam — APPROVED** [review]
LGTM Tested with multiple inline scripts with process exits and CLI errors, the Error message is formatted correctly with exit status and No of steps ran.

**Comment by emilioalvap** [comment]
PostFF LGTM. Tested locally with `8.4`:
image

**Pull request merged**